source/verus/src/main.rs

```diff
@@ -75,19 +63,270 @@ fn main() {
+        std::mem::forget(job);
+    }
+
+    let reports = match reports {
+        Some(reports) => {
+            // proj path is only created after the project has --local-store flag
+            if !reports.proj_path.is_dir() {
+                cmd.status().map_err(|x| format!("verus failed to run with error: {}", x))?;
+                return Ok(());
+            }
+            reports
+        }
+        None => {
+            cmd.status().map_err(|x| format!("verus failed to run with error: {}", x))?;
+            return Ok(());
+        }
+    };
+
+    let proj_path = reports.proj_path;
+
+    cmd.arg(format!("--emit=dep-info={}", reports.dep_path.display()));
+
+    // clean all files in proj_path
+    remove_dir_all(&proj_path).expect("failed to remove repo_path");
+
+    let toml_path = proj_path.join("reports.toml");
+
+    create_dir_all(&proj_path).expect("creating reports.toml");
+    let mut file = File::create(toml_path).expect("creating reports.toml");
+
+    let mut child = cmd
+        .stdout(std::process::Stdio::piped())
+        .stderr(std::process::Stdio::piped())
+        .spawn()
+        .map_err(|x| format!("verus failed to run with error: {}", x))?;
+
+    let out = BufReader::new(child.stdout.take().expect("create BufReader for stdout"));
+    let err = BufReader::new(child.stderr.take().expect("create BufReader for stderr"));
+
+    let thread_err = std::thread::spawn(move || {
+        let mut stderr_output = String::new();
+        err.lines().for_each(|line| {
+            let line = line.unwrap();
+            eprintln!("{}", line);
+            let x = anstream::adapter::strip_str(&line).to_string() + "\n";
+            stderr_output.push_str(&x);
+        });
+        return stderr_output;
+    });
+
+    let thread_out = std::thread::spawn(move || {
+        let mut stdout_output = String::new();
+        out.lines().for_each(|line| {
+            let line = line.unwrap();
+            println!("{}", line);
+            let x = anstream::adapter::strip_str(&line).to_string() + "\n";
+            stdout_output.push_str(&x);
+        });
+        return stdout_output;
+    });
+
+    let err_string = thread_err.join().unwrap();
+    let out_string = thread_out.join().unwrap();
+
+    child.wait().map_err(|x| format!("verus failed to run with error: {}", x))?;
+
+    writeln!(file, "{}", err_string)
+        .map_err(|x| format!("failed to write to reports.toml: {}", x))?;
+    writeln!(file, "{}", out_string)
+        .map_err(|x| format!("failed to write to reports.toml: {}", x))?;
+
+    let dep_file_path = reports.dep_path;
+
+    let deps = get_dependencies(&dep_file_path)?;
+
+    let deps_path = deps.iter().map(|x| Path::new(x)).collect::<Vec<_>>();
+
+    // compute common ancester
+    let common = match common_path::common_path_all(deps_path) {
+        Some(common) => common,
+        None => PathBuf::from(""),
+    };
+
+    if deps.clone().into_iter().any(|dep| dep.is_absolute()) {
```

Comment:
This is likely an unnecessary clone:
```suggestion
    if deps.iter().any(|dep| dep.is_absolute()) {
```

But more importantly, I think this portion makes a few assumptions that I am not sure would always hold. Currently, it assumes that if one file is absolute, then all files are absolute (is this the case?). If it _is_ what you've observed so far, then it might be good to add that as an explicit `assert!(...)` in the body of this `if` block.

Reply:
More generally, one way to handle this is by also adding the current working directory to the set of paths you are taking the common path from, so that you can then remove the outer check of whether all/any are absolute, and do a check inside to check each one.